src/user-management/interfaces/authentication-events.interface.ts

```diff
@@ -0,0 +1,27 @@
+type EventStatus =
+  | {
+      status: 'succeeded';
+    }
+  | {
+      status: 'failed';
+      error: {
+        code: string;
+        message: string;
+      };
+    };
+
+export type EmailVerification = {
+  object: 'email_verification';
+  email: string;
+  ipAddress: string | null;
+  userAgent: string | null;
+  userId: string | null;
+} & EventStatus;
```

Comment:
This union will allow to have a precise type checking that `error` will always be defined when `status` is equal to `failed`.

Reply:
as a start i'd recommend being looser with the types. we can always tighten them later if that seems useful. so what do you think about having one type where `error` is optional? it might be useful to take some inspiration from the types stripe defines in their sdk.

Reply:
yep

Reply:
Makes sense to me to leave it a bit looser, although I'd really love to get the exact states - with the above would mean that the error could be defined as `succeed` which is a bit weird but not a huge friction for the developer's app.

Let me look at how Stripe declares those types -- although I think they are auto-generated, right? So I would expect that it's a bit loose as well.

Reply:
> although I think they are auto-generated

Which is what the future of this SDK will look like as well 😅

Reply:
🤞 lol

Reply:
here's an example: https://github.com/stripe/stripe-node/blob/7293b69a204c4a25ca00574078f6b6699936b3cf/types/Charges.d.ts#L107

Reply:
@amygdalama I don't think Stripe defines event errors on their API - from what I can see [here](https://github.com/stripe/stripe-node/blob/7293b69a204c4a25ca00574078f6b6699936b3cf/types/Events.d.ts#L305).